File: 7_initialization/dotnet/tests/ExampleTests.cs
using NUnit.Framework;
using tests;

namespace Tests
{
    public class Tests
    {
        [Test]
        public void testingObjInit()
        {
            SimplePair pair = new SimplePair
                {
                    Key = 1,
                    Value = "first"
                };

            Assert.AreEqual("first", pair.Value);
        }

        [Test]
        public void testingConstructorCalls() 
        {
            OtherConstructors otherConstructors = new OtherConstructors();
            Assert.AreEqual("default value", otherConstructors.SomeString);
        }
    }
}

File: 7_initialization/javatests/src/test/java/javatests/ExampleTests.java
package javatests;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ExampleTests {

    @Test
    public void testDoubleBrace() {
        SimplePair pair = new SimplePair() {{
            setKey(1);
            setValue("first");
        }};

        assertEquals("first", pair.getValue());
    }

    @Test
    public void testBuilder() {
        SimplePairWithBuilder pair = SimplePairWithBuilder.builder()
                .key(1)
                .value("first")
                .build();

        assertEquals("first", pair.getValue());
    }

    @Test
    public void testConstructorCalls() {
        OtherConstructors constructors = new OtherConstructors();
        assertEquals("default value", constructors.getSomeString());
    }
}
